include/helib/query.h

```diff
@@ -498,13 +555,18 @@ class QueryBuilder
             x.insert(x.end(), j.begin(), j.end());
             prod.push_back(std::move(x));
           }
-
         convertStack.push(std::move(prod));
+      } else if (symbol == "!") {
+        vecvec top = convertStack.top();
+        vecvec clause = negate(top); // negate top of stack
+        convertStack.pop();          // pop
+        convertStack.push(clause);   // push negation
       } else {
-        // Assume it is a number. But sanity check anyway.
+        // Operand: should be a number
         assertTrue(isNumber(symbol),
                    "String is not a number: '" + symbol + "'");
-        convertStack.emplace(vecvec(1, {std::stol(symbol)}));
+        convertStack.emplace(
+            vecvec(1, {std::stol(symbol) + 1})); // using 1 ordering temporarily
```

Comment:
To allow negatives to correspond to nots of columns even when there is a zero column, column i is temporarily written as (i+1): I tried to indicate in the function descriptions and code comments where this was the case.
In particular, in this line, pushing back {{i + 1}} rather than {{i}}.

Reply:
I would prefer the reply text you just wrote as a comment. Verbose, but it makes more sense to anyone coming back to the code

Reply:
I can add something: I mention this on line 400 (in build()) and in the function description of expandOr(). Perhaps given these two I can take the comment out of line 569 instead?